internal/client.go

```diff
@@ -1003,6 +1034,47 @@ func DialCloudOperationsClient(ctx context.Context, options CloudOperationsClien
 	}, nil
 }
 
+// NewUpdateWorkflowOperation returns an UpdateWorkflowOperation that can be used to perform Update-with-Start,
+// or an error in case the update operation is invalid.
+func NewUpdateWorkflowOperation(options UpdateWorkflowOptions) (*UpdateWorkflowOperation, error) {
+	input, err := createUpdateWorkflowInput(options)
+	if err != nil {
+		return nil, err
+	}
+	if options.RunID != "" {
+		return nil, errors.New("RunID is not allowed to be used on an UpdateWorkflowOperation")
+	}
+	if options.FirstExecutionRunID != "" {
+		return nil, errors.New("FirstExecutionRunID is not allowed to be used on an UpdateWorkflowOperation")
+	}
```

Comment:
These error messages could be more helpful.  The issue here seems to be that it doesn't make sense to pass in a run ID for a workflow update for which the workflow does not yet exist, in which case, say that rather than making a very literal code-oriented error message.
Also, if my reasoning is correct, these validations seem more appropriate in the start workflow call, where we actually know that's how this operation will be used.